src/jit/compiler.h

```diff
@@ -1771,7 +1773,11 @@ class Compiler
     // a PSPSym for functions with any EH.
     bool ehNeedsPSPSym() const
     {
+#ifdef _TARGET_X86_
```

Comment:
The main question is whether the unwinder on Linux/x86 is guaranteed to restore the EBP of the parent frame, such that it doesn't need to be restored by the code itself. That is the main reason for the PSPSym: we can't trust that EBP (or the frame pointer, on other architectures) is restored, especially in nested EH scenarios. CoreRT does this properly, and thus doesn't need the PSPSym.

---
In reply to: [95909398](https://github.com/dotnet/coreclr/pull/8889#discussion_r95909398) [](ancestors = 95909398)